README.md

```diff
@@ -35,59 +42,119 @@ There are two Tasks included in this module, `splunk_hec:bolt_apply` and `splunk
+  * Add the `splunk_hec` class.
+  * Configure the following parameters:
+
         ```
         enable_reports = true
         manage_routes = true
-        splunk_token = something like F5129FC8-7272-442B-983C-203F013C1948
+      splunk_token = <TOKEN GENERATED IN STEP 2>
         url = something like https://splunk-8.splunk.internal:8088/services/collector
         include_api_collection = true
         ```
-    5. Hit save
-    6. Run Puppet on the node group, this will cause a restart of the Puppet-Server service
-4. Configure the Splunk Puppet Report Viewer with your HEC token like so\
-![Puppet Report Viewer config](https://raw.githubusercontent.com/puppetlabs/puppetlabs-splunk_hec/v0.8.1/docs/images/puppet_report_viewer_config.png)
-5. Log into the Splunk Console, search `index=* sourcetype=puppet:summary` and if everything was done properly, you should see the reports (and soon facts) from the systems in your Puppet environment
+
+  * Commit the changes.
+  * Run Puppet on the node group; this will cause a restart of the `pe-puppetserver` service.
+
+4. Log into the Splunk console and search `index=* sourcetype=puppet:summary`, if everything was done properly you should see the reports (and soon facts) from the systems in your Puppet environment.
+
+## Custom Installation
+
+> **Please Note**: If you are installing this module using a `control-repo`, you must have `splunk_hec` in your production environment's `Puppetfile` so the Puppet Server process can properly load the required libraries. You can then create a feature branch to enable them and test the configuration, but the libraries **must be** in `production`; otherwise the feature branch won't work as expected. If your Puppet Server is in a different environment, please add this module to the `Puppetfile` in that environment as well.
+
+The steps below will help install and troubleshoot the report processor on a standard Puppet Primary Server; including manual steps to configure compilers (Puppet Servers), and to use the included `splunk_hec` class. Because one is modifying production machines, these steps allow you to validate your settings before deploying the changes live. 
+
+1. Install the Puppet Report Viewer app in Splunk. This will import the needed source types to configure Splunk's HTTP Endpoint Collector (HEC) and provide a dashboard that will show the reports once they are sent to Splunk.
+
+2. Create a Splunk HEC Token or use an existing one that sends to `main` index and **does not** have acknowledgement enabled. Follow the steps provided by Splunk's [Getting Data In Guide](http://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector) if you are new to HTTP Endpoint Collectors.
+
+3. Install this Puppet module in the environment that your Puppet Servers are using (e.g. `production`).
+
+4. Run `puppet plugin download` on your Puppet Servers to sync the content. Some users with strict permissions may need to run `umask 022` first.
```

Comment:
I think it would be good to add a note below this step describing the example error that @Magisus mentioned.

Reply:
Error messaged and note added.